js/reducers/user.js

```diff
@@ -20,9 +20,15 @@ import type { ApiUserProfile } from '../utils/api'
 import config from '../config'
 
 // user state when logged out
-export type LoggedOutUserState = {
+export type DefaultLoggedOutUserState = {
   isLoggedIn: false,
-  loginError?: boolean,
+  apiUrlPrefix: string
+}
+
+// user state when there is an error, tracing by isError=true
+export type ErrorLoggedOutUserState = {
```

Comment:
to make a [disjoint union](https://flow.org/en/docs/types/unions/#toc-disjoint-unions) work you need a discriminating attribute, in this case the discriminator between `DefaultLoggedOutUserState` and `ErrorLoggedOutUserState` is the value of `isError` - what does it mean? it means that Flow, to be able to discriminate between `DefaultLoggedOutUserState` and `ErrorLoggedOutUserState` needs to know which is which based on the value of `isError`, and you tell flow by defining types with actual values, e.g.:

``` js
type DefaultLoggedOutUserState = {
  isError: false
}

type ErrorLoggedOutUserState = {
  isError: true
}
```

Reply:
Fixed.

Reply:
how is it fixed? is the following a valid state/type?

``` js
type ErrorLoggedOutUserState = {
  isError: false
}
```

@fpersico @lussoluca what do you think?

Reply:
The updated code is
`
// user state when logged out
export type DefaultLoggedOutUserState = {
  isLoggedIn: false,
  apiUrlPrefix: string
}

// user state when there is an error, tracing by isError=true
export type ErrorLoggedOutUserState = {
  isLoggedIn: false,
  isError: true,
  apiUrlPrefix: string
}
`

DefaultLoggedOutUserState still need the `isError: false` to make disjoint union work

Reply:
Fixed DefaultLoggedOutUserState with isError: false